### PR TITLE
Add history inspect table

### PR DESCRIPTION
**Pull request by ndrluis**
Reference: #511

### CODE REVIEW

File: pyiceberg/table/__init__.py [hunk 1]
```diff
@@ -3827,6 +3828,38 @@ def _partition_summaries_to_rows(
...
+        import pyarrow as pa
+
+        history_schema = pa.schema([
+            pa.field("made_current_at", pa.timestamp(unit="ms"), nullable=False),
+            pa.field("snapshot_id", pa.int64(), nullable=False),
+            pa.field("parent_id", pa.int64(), nullable=True),
+            pa.field("is_current_ancestor", pa.bool_(), nullable=False),
+        ])
+
+        ancestor_ids = set(
```
Comment (kevinjqliu) [inline pyiceberg/table/__init__.py:3841]: nit: I think the helper function can be replaced with a 1-liner

```
{snapshot.snapshot_id for snapshot in ancestors_of(self.tbl.current_snapshot(), self.tbl.metadata)}
```

**Review by kevinjqliu — COMMENTED** [review]
lgtm! added some minor nits

**Review by kevinjqliu — APPROVED** [review]
thanks! LGTM

**Review by Fokko — APPROVED** [review]
Awesome! Thanks @ndrluis for working on this, and thanks @kevinjqliu for the review 🙌

**Pull request merged**